conda-store-server/conda_store_server/action/base.py

```diff
@@ -38,20 +38,23 @@ class ActionContext:
     def __init__(self):
         self.id = str(uuid.uuid4())
         self.stdout = io.StringIO()
+        self.stderr = io.StringIO()
         self.log = logging.getLogger(f"conda_store_server.action.{self.id}")
         self.log.propagate = False
         self.log.addHandler(logging.StreamHandler(stream=self.stdout))
         self.log.setLevel(logging.INFO)
         self.result = None
         self.artifacts = {}
 
-    def run(self, *args, **kwargs):
+    def run(self, *args, redirect_stderr=True, **kwargs):
         result = subprocess.run(
             *args,
             **kwargs,
             stdout=subprocess.PIPE,
-            stderr=subprocess.STDOUT,
+            stderr=subprocess.STDOUT if redirect_stderr else subprocess.PIPE,
             encoding="utf-8",
         )
         self.stdout.write(result.stdout)
+        if not redirect_stderr:
+            self.stderr.write(result.stderr)
         return result
```

Comment:
See https://github.com/conda-incubator/conda-store/pull/559#issuecomment-1756298903 for context on redirects.